mltsp/Flask/flask_app.py

```diff
@@ -681,6 +659,86 @@ def list_featuresets(
         return authed_featuresets
 
 
+# TODO database module
+def list_datasets_cursor_to_html_table(cursor):
+    """
+    """
+    authed_datasets = (
+        "<table id='datasets_table' style='display:none;'>" +
+        "   <tr class='datasets_row'>" +
+        "       <th>Dataset set name</th>" +
+        "       <th>Date created</th>" +
+        "       <th>Number of files</th>" +
+        "       <th>Remove from database</th>" +
+        "   </tr>")
+    count = 0
+    for entry in cursor:
+        authed_datasets += (
+            "<tr class='datasets_row'><td align='left'>" +
+            entry['name'] + "</td><td align='left'>" +
+            entry['created'][:-13] +
+            "<td align='center'>" + str(len(entry["ts_filenames"])) + "</td>" +
+            ("<td align='center'><input "
+             "type='checkbox' name='delete_dataset_key' "
+             "value='%s'></td></tr>") % entry['id'])
+        count += 1
+    authed_datasets += "</table>"
+    return authed_datasets
+
+
+def list_datasets(
+        auth_only=True, by_project=False, name_only=False,
+        as_html_table_string=False):
+    """Return list of strings describing entries in 'datasets' table.
+
+    Parameters
+    ----------
+    auth_only : bool, optional
```

Comment:
All these options seem extraneous.  Do we ever need to query all the datasets in existence (i.e., can auth_only go)? Why not always return a dictionary of ``{'project': [dataset1, dataset2]}``, then ``by_project`` can go.  And why not always include the dates in the result?  And certainly never return as an HTML string.